### PR TITLE
Scorecard e2e Testing

### PR DESCRIPTION
**Pull request by theishshah**
Add basic and olm scorecard tests to e2e test suite.

### CODE REVIEW

File: test/e2e/e2e_suite.go [hunk 1]
```diff
@@ -211,6 +213,33 @@ var _ = Describe("operator-sdk", func() {
...
+			By("running basic scorecard tests")
+			var scorecardOutput v1alpha3.TestList
+			runScorecardCmd := exec.Command(tc.BinaryName, "scorecard", "bundle",
+				"--selector=suite=basic",
+				"--output=json",
+				"--skip-cleanup=true",
+				"--wait-time=40s")
+			scorecardOutputBytes, err := tc.Run(runScorecardCmd)
+			Expect(err).NotTo(HaveOccurred())
+			err = json.Unmarshal(scorecardOutputBytes, &scorecardOutput)
```
Comment (estroz) [inline test/e2e/e2e_suite.go:226]: This is so nice to be able to do.

File: test/e2e/e2e_suite.go [hunk 2]
```diff
@@ -211,6 +213,33 @@ var _ = Describe("operator-sdk", func() {
 				"--timeout", "4m")
 			_, err = tc.Run(cleanupPkgManCmd)
 			Expect(err).NotTo(HaveOccurred())
+
+			By("running basic scorecard tests")
```
Comment (estroz) [inline test/e2e/e2e_suite.go:217]: You will have to run these steps before `cleanup packagemanifests` is called.

File: test/e2e/e2e_suite.go [hunk 3]
```diff
@@ -204,6 +206,42 @@ var _ = Describe("operator-sdk", func() {
...
+			Expect(err).To(HaveOccurred())
+			err = json.Unmarshal(scorecardOutputBytes, &scorecardOutput)
+			Expect(err).NotTo(HaveOccurred())
+
+			resultTable := make(map[string]v1alpha3.State)
+			resultTable["olm-status-descriptors"] = v1alpha3.FailState
+			resultTable["olm-crds-have-resources"] = v1alpha3.FailState
+			resultTable["olm-bundle-validation"] = v1alpha3.PassState
+			resultTable["olm-spec-descriptors"] = v1alpha3.FailState
+			resultTable["olm-crds-have-validation"] = v1alpha3.PassState
```
Comment (estroz) [inline test/e2e/e2e_suite.go:239]: Use these [constants](https://github.com/operator-framework/operator-sdk/blob/master/internal/scorecard/tests/olm.go#L40-L44).

File: test/e2e/e2e_suite.go [hunk 4]
```diff
@@ -204,6 +206,42 @@ var _ = Describe("operator-sdk", func() {
...
+
+			for a := 0; a < len(scorecardOutput.Items); a++ {
```
Comment (estroz) [inline test/e2e/e2e_suite.go:241]: Use test names to index into the `resultTable` map to make sure each test is run.

File: test/e2e/e2e_suite.go [hunk 5]
```diff
@@ -204,6 +206,42 @@ var _ = Describe("operator-sdk", func() {
...
+			err = json.Unmarshal(scorecardOutputBytes, &scorecardOutput)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(len(scorecardOutput.Items)).To(Equal(1))
+			Expect(scorecardOutput.Items[0].Status.Results[0].State).To(Equal(v1alpha3.PassState))
+
+			By("running olm scorecard tests")
+			runOLMScorecardCmd := exec.Command(tc.BinaryName, "scorecard", "bundle",
+				"--selector=suite=olm",
+				"--output=json",
+				"--skip-cleanup=true",
```
Comment (estroz) [inline test/e2e/e2e_suite.go:227]: Any reason we're skipping cleanup?

File: test/e2e/e2e_suite.go (hunk 2 above)
Comment (camilamacedo86) [reply test/e2e/e2e_suite.go:209]: It shows done closing it as solved.

**Pull request merged**